src/utils.rs

```diff
@@ -0,0 +1,46 @@
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use std::cmp::Ordering;
+use log::{info, warn, debug, error};
+
+pub const K: usize = 20; // Maximum number of nodes in a k-bucket
+pub const ALPHA: usize = 3; // Number of parallel lookups
+pub const BOOTSTRAP_NODES: [&str; 1] = ["127.0.0.1:33333"]; // Hardcoded bootstrap node
+
+
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize, Debug)]
+pub struct NodeId([u8; 32]);
```

Comment:
you might want to use hex encoding to more conveniently represent this for `Display` / `Debug`.